### PR TITLE
Making setup.py install work for Caffe2

### PR DESCRIPTION
**Pull request by pjh5**
Tested on my mac on a pretty clean anaconda3

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -548,6 +545,13 @@ def build_extensions(self):
                     i += 1
         distutils.command.build_ext.build_ext.build_extensions(self)
 
+    def get_outputs(self):
+        outputs = distutils.command.build_ext.build_ext.get_outputs(self)
+        if not FULL_CAFFE2:
```
Comment (orionr) [inline setup.py]: Nit - can you make this

```
outputs = distutils.command.build_ext.build_ext.get_outputs(self)
if FULL_CAFFE2:
  outputs += [os.path.join(self.build_lib, d) for d in ['caffe', 'caffe2']]
return outputs
```

**Review by orionr — APPROVED** [review]

**Review by soumith — APPROVED** [review]

**Comment by pjh5** [comment]
Erm, so actually it works but not after a git clean -xfd cause caffe2_pybind11_state still has a hardcoded rpath to the build folder.

**Comment by ezyang** [comment]
Hi, what's the status on this?

**Comment by orionr** [comment]
As a part of this, can we remove `setup_caffe2.py`? We'll need to update docs to switch to`FULL_CAFFE2=1 python setup.py develop/install`, but might be time.

**Comment by pjh5** [comment]
Right now going through setup.py installs libcaffe2.so in python_root/lib/pythonx.x/site-packages/torch/lib/libcaffe2.so but going through our old cmake+make path (not going through build_torch_libs.sh) installs into python_root/lib/libcaffe2.so , so the rpath of caffe2_pybind11_state.so needs to depend on the install path

The right thing to do should be making python_root/lib/libcaffe2.so always the install location, so that you can e.g. install just Caffe2 and then later install Pytorch without having multiple libcaffe2.sos

**Comment by pjh5** [comment]
@orionr this is ready.

**Comment by goldsborough** [comment]
This PR didn't pass the flake8 lint

**Pull request closed without merging**